src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -447,7 +447,7 @@ static int add_esp(DissectedPartition *p) {
         r = fstab_is_mount_point(esp);
         if (r < 0)
                 return log_error_errno(r, "Failed to parse fstab: %m");
-        if (r == 0) {
+        if (r == true) {
```

Comment:
ouch! indeed!

Hmm, but please do no compare integers with "true" and "false", just as a matter of coding style. Comparing the C99 "bool" type with "true" and "false" is fine, but if we operate on "int", then please don't the reason for that is that "int" can take more than 0 or 1, and traditionally all of > 0 was considered conceptually true, not just 1. Or to say this differently:

```c
int x = 7;
if (x == true)
        printf("YES\n");
```

and this:
```c
bool x = 7;
if (x == true)
        printf("YES\n");
```

yield very different results.

While hence comparing an "int" with "true" or "false" is not OK, assigning "true" or "false" to an int is fine though, as in that case it just degrades to "1" and "0"... 

Yes, that's C!